dds/DCPS/DataWriterImpl.cpp

```diff
@@ -1446,14 +1447,26 @@ DataWriterImpl::enable()
 
   publisher->get_qos(pub_qos);
 
+  TypeSupportImpl* const typesupport =
+      dynamic_cast<TypeSupportImpl*>(topic_servant_->get_type_support());
+  size_t sto = XTypes::find_size(typesupport->getMinimalTypeObject());
+  XTypes::TypeIdentifierPtr type_iden = XTypes::makeTypeIdentifier(typesupport->getMinimalTypeObject());
+  XTypes::TypeInformation type_info;
+  type_info.minimal.typeid_with_size.type_id = type_iden;
+  type_info.minimal.typeid_with_size.typeobject_serialized_size = sto;
+  type_info.minimal.dependent_typeid_count = 0;
+  type_info.complete.dependent_typeid_count = 0;
+
   this->publication_id_ =
     disco->add_publication(this->domain_id_,
                            this->dp_id_,
                            this->topic_servant_->get_id(),
                            this,
                            this->qos_,
                            trans_conf_info,
-                           pub_qos);
+                           pub_qos,
+                           type_info,
+                           swap_bytes());
```

Comment:
I don't think this use of `swap_bytes` is right, let's discuss it tomorrow.

Reply:
The Xtypes spec says that byte ordering of the octet in the parameter should be the same as in the message.

Reply:
There are different messages though.  The ParameterList is part of discovery (SEDP), the DataWriter's swap bytes is for the samples it will send to the transport.